index.js

```diff
@@ -1,141 +1,143 @@
       }
-    }
 
-    const getElementFromCoords = (coords) => {
-      const row = elements[coords.y]
-      if (row) {
-        const element = row[coords.x]
-        return element
+      const getIntAttr = ({element, attr}) => { return parseInt(element.getAttribute(attr)) } // external functions: done, global functions parseInt, getAttribute
+
+      const getCoordsFromElement = ({ target, getIntAttr }) => { // external functions: non global: getIntAttr
+        const element = target;
+        const columnIndex = getIntAttr({element: element, attr: 'data-columnIndex'});
+        const rowIndex = getIntAttr({element: element, attr: 'data-rowIndex'})
+        return { x: columnIndex, y: rowIndex } }
+
+      const getAdjacentCells = ({coords, getElementFromCoords}) => { // external functions: non global: getElementFromCoords, global: filter
+        const x = coords.x; const y = coords.y; const n = y - 1;
+        const s = y + 1; const wx = x - 1; const ex = x + 1;
+        const ny = x % 2 === 1 ? y : n; const sy = x % 2 === 1 ? s : y;
+        // console.log(elements) // global value.
+        return [
+          getElementFromCoords({  x: x, y: n , elements: elements }),   // N
+          getElementFromCoords({  x: x, y: s , elements: elements }),   // S
+          getElementFromCoords({  x: wx, y: ny , elements: elements }), // NW
+          getElementFromCoords({  x: ex, y: ny , elements: elements }), // NE
+          getElementFromCoords({  x: wx, y: sy , elements: elements }), // SW
+          getElementFromCoords({  x: ex, y: sy , elements: elements }), // SE
+        ].filter(x => !!x)
       }
-    }
 
-    const getAdjacentCells = (coords) => {
-      const x = coords.x
-      const y = coords.y
-      const n = y - 1
-      const s = y + 1
-      const wx = x - 1
-      const ex = x + 1
-      const ny = x % 2 === 1 ? y : n
-      const sy = x % 2 === 1 ? s : y
-      return [
-        // N
-        getElementFromCoords({ x: x, y: n }),
-        // S
-        getElementFromCoords({ x: x, y: s }),
-        // NW
-        getElementFromCoords({ x: wx, y: ny }),
-        // NE
-        getElementFromCoords({ x: ex, y: ny }),
-        // SW
-        getElementFromCoords({ x: wx, y: sy }),
-        // SE
-        getElementFromCoords({ x: ex, y: sy }),
-      ]
-        .filter(x => !!x)
-    }
+      const handleCellMouseOver = ({target}) => { // external functions: getCoordsFromElement, getAdjacentCells, addClass
+        const coords = getCoordsFromElement({ target, getIntAttr })
+        addClass({ element:target, className:'hover'})
 
-    const addClass = (element, className) => {
-      if (element.className.indexOf(className) === -1) {
-        element.className += (' ' + className)
+      const adjacentCells = getAdjacentCells({coords: coords, getElementFromCoords})
+        adjacentCells.forEach(cell => {
+          addClass({ element:cell, className:'adjacent' })
+        })
+      }
```

Comment:
For the above function: handleCellMouseOver
I could not figure out how to pass, functions to this method with the .bind operator.

open to suggestions.